bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java

```diff
@@ -0,0 +1,433 @@
+                case REWIND:
+                    logger.warn("Not implemented yet");
+                    break;
+            }
+        } else if (command instanceof RefreshType) {
+            volumio.getState();
+        } else {
+            logger.error("Command is not handled: {}", command);
+        }
+    }
+
+    /**
+     * Bind default listeners to volumio session.
+     * - EVENT_CONNECT - Connection to volumio was established
+     * - EVENT_DISCONNECT - Connection was disconnected
+     * - PUSH.STATE -
+     */
+    private void bindDefaultListener() {
+        volumio.on(Socket.EVENT_CONNECT, connectListener());
+        volumio.on(Socket.EVENT_DISCONNECT, disconnectListener());
+        volumio.on(VolumioEvents.PUSH_STATE, pushStateListener());
+    }
+
+    /**
+     * Read the configuration and connect to volumio device. The Volumio impl. is
+     * async so it should not block the process in any way.
+     */
+    @Override
+    public void initialize() {
+        String hostname = (String) getThing().getConfiguration().get(VolumioBindingConstants.CONFIG_PROPERTY_HOSTNAME);
+        int port = ((BigDecimal) getThing().getConfiguration().get(VolumioBindingConstants.CONFIG_PROPERTY_PORT))
+                .intValueExact();
+        String protocol = (String) getThing().getConfiguration().get(VolumioBindingConstants.CONFIG_PROPERTY_PROTOCOL);
+        int timeout = ((BigDecimal) getThing().getConfiguration().get(VolumioBindingConstants.CONFIG_PROPERTY_TIMEOUT))
+                .intValueExact();
+
+        if (hostname == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Configuration incomplete, missing hostname");
+        } else if (protocol == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Configuration incomplete, missing protocol");
+        } else {
+            logger.debug("Trying to connect to Volumio on {}://{}:{}", protocol, hostname, port);
+            try {
+                volumio = new VolumioService(protocol, hostname, port, timeout);
+                clearChannels();
+                bindDefaultListener();
+                updateStatus(ThingStatus.OFFLINE);
+                volumio.connect();
+            } catch (Exception e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (volumio != null) {
+            scheduler.schedule(new Runnable() {
+                @Override
+                public void run() {
+                    if (volumio.isConnected()) {
+                        logger.warn("Timeout during disconnect event");
+                    } else {
+                        volumio.close();
+                    }
+                    clearChannels();
+                }
+            }, 30, TimeUnit.SECONDS);
```

Comment:
Use lambdas for runnables to reduce code:

```suggestion
            scheduler.schedule(() -> {
                  if (volumio.isConnected()) {
                      logger.warn("Timeout during disconnect event");
                  } else {
                      volumio.close();
                  }
                  clearChannels();
            }, 30, TimeUnit.SECONDS);
```